src/main/java/org/apache/groovy/ast/tools/ExpressionUtils.java

```diff
@@ -309,8 +309,15 @@ public static Expression transformInlineConstants(final Expression exp) {
             }
         } else if (exp instanceof BinaryExpression) {
             BinaryExpression be = (BinaryExpression) exp;
-            be.setLeftExpression(transformInlineConstants(be.getLeftExpression()));
-            be.setRightExpression(transformInlineConstants(be.getRightExpression()));
+            Expression lhs = transformInlineConstants(be.getLeftExpression());
+            Expression rhs = transformInlineConstants(be.getRightExpression());
+            if (lhs instanceof ConstantExpression && rhs instanceof ConstantExpression &&
```

Comment:
We should check `be.operation.type == Types.PLUS` too?

Reply:
Yes, thanks.  I did have that when the code was located elsewhere.  I will get it updated.